modules/generator/processor/hostinfo/processor.go

```diff
@@ -0,0 +1,71 @@
+package hostinfo
+
+import (
+	"context"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/tempo/modules/generator/registry"
+	"github.com/grafana/tempo/pkg/tempopb"
+	"go.opentelemetry.io/otel"
+)
+
+const (
+	Name = "host-info"
+
+	hostInfoMetric     = "traces_host_info"
+	hostIdentifierAttr = "grafana.host.id"
+)
+
+var tracer = otel.Tracer("modules/generator/processor/hostinfo")
+
+type Processor struct {
+	Cfg    Config
+	logger log.Logger
+
+	gauge      registry.Gauge
+	registry   registry.Registry
+	metricName string
+}
+
+func (p *Processor) Name() string {
+	return Name
+}
+
+func (p *Processor) PushSpans(ctx context.Context, req *tempopb.PushSpansRequest) {
+	_, span := tracer.Start(ctx, "hostinfo.PushSpans")
+	defer span.End()
+
+	for i := 0; i < len(req.Batches); i++ {
+		resourceSpan := req.Batches[i]
+		attrs := resourceSpan.GetResource().GetAttributes()
+
+	idAttrLoop:
+		for _, idAttr := range p.Cfg.HostIdentifiers {
+			for _, attr := range attrs {
+				if attr.GetKey() == idAttr {
+					labelValues := p.registry.NewLabelValueCombo(
+						[]string{hostIdentifierAttr},
```

Comment:
both of these string slices can be alloc'ed once and stored on the struct. the first one can be alloc'ed and never changed. the second you would just set the string at index 0 each time before you call NewLabelValueCombo.

Reply:
Thanks for the suggestion. I'll make this update. I haven't looked too closely, but I assume this method could be called concurrently? If so, I can add the necessary synchronization mechanisms for writing into the value slice.

Reply:
that's a good point. rather than adding concurrency mechanisms maybe just alloc these once at the top of the func?

i'm also no wondering if go is smart enough to just put this on the stack and this whole conversation is moot :)

Reply:
sounds good! I added the labels string slice to the struct as that never changes. I alloc'd values slice at the top of the func.